redis/redis.go

```diff
@@ -146,59 +146,114 @@ func (c *Client) PublishEnabled(state *EnabledMessage) {
 		})
 
 		if mErr == nil {
-			c.client.Publish(c.ctx, SyncChannelName, binMsg)
+			c.client.Do(ctx,
+				c.client.B().Publish().
+					Channel(SyncChannelName).
+					Message(rueidis.BinaryString(binMsg)).
+					Build())
 		}
 	}
 }
 
 // GetRedisCache reads the redis cache and publish it to the channel
-func (c *Client) GetRedisCache() {
+func (c *Client) GetRedisCache(ctx context.Context) {
 	c.l.Debug("GetRedisCache")
 
 	go func() {
-		iter := c.client.Scan(c.ctx, 0, prefixKey("*"), 0).Iterator()
-		for iter.Next(c.ctx) {
-			response, err := c.getResponse(iter.Val())
-			if err == nil {
-				if response != nil {
-					c.CacheChannel <- response
-				}
-			} else {
+		var cursor uint64
+
+		searchKey := prefixKey("*")
+
+		for {
+			sres, err := c.client.Do(ctx, c.client.B().Scan().Cursor(cursor).Match(searchKey).Count(1).Build()).AsScanEntry()
+			if err != nil {
 				c.l.Error("GetRedisCache ", err)
```

Comment:
Would be nice to expand the error message, not 100% sure what I wrote is appropriate, so please improve if you wish:
```suggestion
				c.l.Errorf("could not start a searching through Redis cache: %s", err)
```

Reply:
Error description is correct 👍